pay-api/src/pay_api/services/payment_account.py

```diff
@@ -373,11 +373,19 @@ def create(cls, account_request: Dict[str, Any] = None, is_sandbox: bool = False
     @classmethod
     def _check_and_handle_payment_method(cls, account: PaymentAccountModel, target_payment_method: str):
         """Check if the payment method has changed and invoke handling logic."""
-        if account.payment_method == target_payment_method or target_payment_method != PaymentMethod.EFT.value:
+        if account.payment_method == target_payment_method or \
+                PaymentMethod.EFT.value not in {account.payment_method, target_payment_method}:
             return
 
-        # Payment method has changed and is EFT
-        Statement.generate_interim_statement(account.auth_account_id, StatementFrequency.MONTHLY.value)
+        if (account.payment_method == PaymentMethod.EFT.value and
+                target_payment_method != PaymentMethod.EFT.value):
+            Statement.generate_interim_statement(account.auth_account_id,
+                                                 StatementFrequency.PAYMENT_METHOD_CHANGED.value,
```

Comment:
This probably isn't necessary and the value for StatementFrequency.PAYMENT_METHOD_CHANGED isn't really a frequency.

We can probably use the default in enums.py when switching to a new payment method that isn't EFT since we know that is monthly.
`def default_frequency():`

Reply:
Probably can simplify the conditional if that is the case as well

```
statement_frequency = StatementFrequency.MONTHLY.value if account.payment_method == PaymentMethod.EFT.value else StatementFrequency.get_default_frequency

Statement.generate_interim_statement(account.auth_account_id, statement_frequency)

```

Reply:
how about in the statement table? what frequency suppose to save there? it shows "Payment Method Changed" under frequency at UI
![image](https://github.com/bcgov/sbc-pay/assets/16507223/7d9a5147-56e8-4887-960f-81b019661840)

Reply:
Currently a statement record will return the payment methods for the duration of the statement using history records.

https://github.com/bcgov/sbc-pay/blob/main/pay-api/src/pay_api/models/statement.py#L73

and in auth web, if there are multiple payment methods we know it has changed for this statement

https://github.com/bcgov/sbc-auth/blob/main/auth-web/src/components/auth/account-settings/statement/Statements.vue#L319